### DISCUSSION TITLE
Variable Scope?

### DISCUSSION THREAD

**Discussion by flugruger** [nextgenhealthcare/connect#4857] [Q&A] [answered]
Could you help with info about the scope of JavaScript variables in transformers?
If I define a JS variable in step 1, will that be available in later transformer steps?

I've been trying this out, defining my key configuration variables in an initial step. As far as I can tell, these variables are then available throughout the transformer. 

Problem is:  the more experienced Mirth developers say this cannot work. They insist I'll need to assign the variables to the channel map. 

Can you help with what's correct?

Thanks!

**Comment by pacmano1** [comment, marked as answer]
Like your last question, that is simple enough to test and simple enough to point out to the senior folks that a variable defined in transformer step one can be referred in transformer step two.

![image](https://user-images.githubusercontent.com/44065187/141857645-c7bd536a-3835-4b31-baa1-660399f11a6b.png)

![image](https://user-images.githubusercontent.com/44065187/141857676-a1e1b039-a6de-40f3-a0cf-94819e4dcb9f.png)

![image](https://user-images.githubusercontent.com/44065187/141857701-69e1102c-eb68-4e9a-923c-202f15688e3b.png)

**Reply by pacmano1** [reply]
There are likely other reasons your team wants to put stuff in a channel map variable or some other variable.   Including being able to see it in the message viewer.   In fact it is often superior to logger.info because you see the value of the var in the context of the message rather that a stream of errors / info scrolling in the server log.

**Reply by flugruger** [reply]
Thanks, Pacmano1. I've been fighting with some very odd issues the last three days, so needed to have an expert's confirmation.
